test/policies/group_offers_policy_test.rb

```diff
@@ -0,0 +1,25 @@
+require 'test_helper'
+
+class GroupOfferPolicyTest < PolicyAssertions::Test
+  test 'superadmin can use all actions' do
+    assert_permit(create(:user), GroupOffer, 'new?', 'create?', 'index?', 'show?', 'edit?',
+      'update?', 'destroy?')
+  end
+
+  test 'social worker and volunteer have no access' do
+    refute_permit(create(:social_worker), GroupOffer, 'new?', 'create?', 'index?', 'show?',
+      'edit?', 'update?', 'destroy?')
+    refute_permit(create(:user_volunteer), GroupOffer, 'new?', 'create?', 'index?', 'show?',
+      'edit?', 'update?', 'destroy?')
+  end
+
+  test 'department manager has limited access' do
+    department_manager = create :department_manager, :with_departments
+    create :group_offer
```

Comment:
This way you can never be 100% sure to get the right group offer and department manager.
As create returns the object, you can just do it like following, and it will allways be 100% sure:
```ruby
department_manager = create :department_manager, :with_departments
group_offer_first = create :group_offer
group_offer_second = create :group_offer, department: department_manager.department.last
# ...
refute_permit(department_manager, group_offer_first, 'show?', 'edit?', 'update?')
assert_permit(department_manager, group_offer_second, 'show?', 'edit?', 'update?')
```